riscv/runtime/src/allocator.rs

```diff
@@ -0,0 +1,65 @@
+//! A very simple global allocator.
+//!
+//! Allocates on a global array and never deallocates.
+
+use core::{
+    alloc::{GlobalAlloc, Layout},
+    cell::Cell,
+    ptr::{self, addr_of},
+};
+
+struct FixedMemoryAllocator<const SIZE: usize> {
+    mem_buffer: [u8; SIZE],
+    next_available: Cell<usize>,
```

Comment:
Why does this need to be a `Cell` and what are the consequences?

Reply:
It needs `Cell` because `GlobalAlloc::alloc()` trait API takes `&self` instead of `&mut self`. `Cell has no overhead, but no multithread protection either (which isn't a problem for us, I believe).

Reply:
`Cell` is just an access controller, its only purpose is to allow mutating a non-mut object, while preventing taking any king of refs to it.

Reply:
Ok, thanks!